spec/registry.xml

```diff
@@ -0,0 +1,250 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-model href="registry.dtd" type="application/xml-dtd"?>
+<registry>
+  <pattern id="anyNumber" regex="-?[0-9]+(\.[0-9]+)"/>
+  <pattern id="positiveInteger" regex="[0-9]+"/>
+  <pattern id="currencyCode" regex="[A-Z]{3}"/>
+  <pattern id="timeZoneId" regex="[a-zA-Z/]+"/>
+  <pattern id="anythingNotEmpty" regex=".+"/>
+
+  <!-- I don't think that regexp is powerfull enough to
+    describe all things we might want, especially for input.
+    ISO 8601 is one example.
+    https://en.wikipedia.org/wiki/ISO_8601
+    So for now this is a very "sketchy" regexp.
+
+    One can also imagine phone numbers, emails, url, etc.
+
+    Even the regex for anyNumber (from the .md) is not OK:
+      .3  : Invalid (no digit before dot)
+      3   : Invalid (no dot + one digit after)
+            Brackets in regexp means grouping, not optional.
+      002 : Valid (OK to start with zero)
+      000 : Valid (OK to start with zero)
+  -->
+  <pattern id="iso8601" regex="\d{4}-\d{2}-\d{2}T\d{2}:\d{2}\d{2}"/>
```

Comment:
ISO8601 and RFC3339 are in the process of being extended and that work is important for support of Temporals. So we probably want to revisit here.

Reply:
Agree. I that what we have now (regex + list of values) is not powerful enough.
Created https://github.com/unicode-org/message-format-wg/issues/422 for this (and more)